packages/aws-cdk-lib/aws-ecr/lib/lifecycle.ts

```diff
@@ -96,3 +96,150 @@ export enum TagStatus {
    */
   UNTAGGED = 'untagged',
 }
+
+/**
+ * Properties for creating a LifecycleRule
+ */
+export interface LifecycleRuleProps extends LifecycleRule {}
+
+/**
+ * An ECR lifecycle rule class that provides JSON serialization capability
+ *
+ * This class provides the same properties as the LifecycleRule interface and adds a toJSON() method
+ * for use with L1 constructs that require JSON serialization of lifecycle rules.
+ *
+ * @example
+ * // For use with L1 constructs
+ * declare const stack: Stack;
+ *
+ * const lifecycleRule = new ecr.LifecycleRuleClass({
+ *   maxImageCount: 5,
+ *   tagStatus: ecr.TagStatus.TAGGED,
+ *   tagPrefixList: ['prod']
+ * });
+ *
+ * const template = new ecr.CfnRepositoryCreationTemplate(stack, 'Template', {
+ *   appliedFor: ['PULL_THROUGH_CACHE'],
+ *   prefix: 'my-prefix',
+ *   lifecyclePolicy: JSON.stringify({
+ *     rules: [lifecycleRule.toJSON()]
+ *   })
+ * });
+ */
+export class LifecycleRuleClass {
+  /**
+   * Controls the order in which rules are evaluated (low to high)
+   */
+  public readonly rulePriority?: number;
```

Comment:
I don't really like exposing the members of the input props on the class. It ties us into a contract that limits future flexibility (for example, accepting unions as inputs), typically for little benefit.